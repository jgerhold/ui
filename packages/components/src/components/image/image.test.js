import { mount } from '@vue/test-utils'
import Image from './Image'

describe('Image', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(Image)
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(Image)
      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'url',
        value: 'http://foo.bar'
      },
      {
        prop: 'alt',
        value: 'alternative title'
      },
      {
        prop: 'color',
        value: '#fff'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(Image, {
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
