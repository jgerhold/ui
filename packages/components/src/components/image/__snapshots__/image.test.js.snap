// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image props should render with prop alt 1`] = `
<div
  class="image-container"
>
  <div
    class="cover"
    name="fade"
    style="background-color: rgb(43, 138, 198);"
  />
   
  <!---->
</div>
`;

exports[`Image props should render with prop color 1`] = `
<div
  class="image-container"
>
  <div
    class="cover"
    name="fade"
    style="background-color: rgb(255, 255, 255);"
  />
   
  <!---->
</div>
`;

exports[`Image props should render with prop url 1`] = `
<div
  class="image-container"
>
  <div
    class="cover"
    name="fade"
    style="background-color: rgb(43, 138, 198);"
  />
   
  <img
    class="image"
    src="http://foo.bar"
    style="height: 0px;"
  />
</div>
`;

exports[`Image props should render without props 1`] = `
<div
  class="image-container"
>
  <div
    class="cover"
    name="fade"
    style="background-color: rgb(43, 138, 198);"
  />
   
  <!---->
</div>
`;
